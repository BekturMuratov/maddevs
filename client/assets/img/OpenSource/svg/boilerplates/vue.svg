<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>