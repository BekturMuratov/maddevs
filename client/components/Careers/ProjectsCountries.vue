<template>
  <div class="projects-countries container">
    <h2 class="projects-countries_title">
      Global projects in 20+ countries
    </h2>
    <p class="projects-countries_description">
      We build solutions for 35M+ users worldwide
    </p>
    <div class="projects-countries_img-wrap">
      <img
        :data-src="$getMediaFromS3('/images/Careers/svg/сountries.svg')"
        alt="lobal projects in 20+ countries"
        width="1240"
        height="558"
        class="projects-countries_img media_lazy"
      >
    </div>
    <UICustomersList
      class="projects-countries_customers-list"
      :customers="careersCustomers"
    />
    <UILinkButton
      class="projects-countries_link-btn"
      link="/case-studies/"
      title="See all case studies "
    />
  </div>
</template>

<script>
import UICustomersList from '@/components/shared/UICustomersList'
import UILinkButton from '@/components/shared/UILinkButton'
import { careersCustomers } from '@/data/customers'

export default {
  name: 'ProjectsCountries',
  components: {
    UICustomersList,
    UILinkButton,
  },

  data() {
    return {
      careersCustomers,
    }
  },
}
</script>

<style lang="scss" scoped>
@import '@/assets/styles/_vars';

.projects-countries {
  margin-bottom: 233px;

  &_title,
  &_description {
    letter-spacing: -0.04em;
    line-height: 72px;
    text-align: center;
    color: $text-color--white;
  }

  &_title {
    @include font('Inter', 60px, 800);
  }

  &_description {
    margin-bottom: 9px;
    @include font('Inter', 20px, 400);
  }

  &_img {
    width: 100%;
    height: 100%;
  }

  &_customers-list {
    margin-bottom: 40px;
  }

  &_link-btn {
    margin: 0 auto;
  }
}
</style>
