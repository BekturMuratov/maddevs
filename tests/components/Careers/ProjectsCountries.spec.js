import ProjectsCountries from '@/components/Careers/ProjectsCountries'
import { render } from '@testing-library/vue'

const mocks = {
  $getMediaFromS3: () => 'img.jpg',
  $t: () => 'translated',
}

const stubs = ['UICustomersList', 'UILinkButton']

const directives = {
  'lazy-load': () => {},
}

describe('ProjectsCountries component', () => {
  it('should render correctly', () => {
    const { container } = render(ProjectsCountries, {
      mocks,
      stubs,
      directives,
    })

    expect(container).toMatchSnapshot()
  })
})
