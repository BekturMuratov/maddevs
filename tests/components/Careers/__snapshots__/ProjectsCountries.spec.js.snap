// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsCountries component should render correctly 1`] = `
<div>
  <div
    class="projects-countries container"
  >
    <h2
      class="projects-countries_title"
    >
      
    Global projects in 20+ countries
  
    </h2>
     
    <p
      class="projects-countries_description"
    >
      
    We build solutions for 35M+ users worldwide
  
    </p>
     
    <div
      class="projects-countries_img-wrap"
    >
      <img
        alt="lobal projects in 20+ countries"
        class="projects-countries_img media_lazy"
        data-src="img.jpg"
        height="558"
        width="1240"
      />
    </div>
     
    <div
      class="customers__customers-list projects-countries_customers-list"
    >
      <div
        class="icon-wrapper"
        data-testid="icon-wrapper"
        style="background-color: transparent;"
      >
        <img
          alt="clutch"
          class="icon-wrapper__icon media_lazy"
          data-src="[object Object]"
          data-testid="customer-icon"
        />
      </div>
      <div
        class="icon-wrapper"
        data-testid="icon-wrapper"
        style="background-color: transparent;"
      >
        <img
          alt="godee"
          class="icon-wrapper__icon media_lazy"
          data-src="[object Object]"
          data-testid="customer-icon"
        />
      </div>
      <div
        class="icon-wrapper"
        data-testid="icon-wrapper"
        style="background-color: transparent;"
      >
        <img
          alt="sjmc"
          class="icon-wrapper__icon media_lazy"
          data-src="[object Object]"
          data-testid="customer-icon"
        />
      </div>
      <div
        class="icon-wrapper"
        data-testid="icon-wrapper"
        style="background-color: transparent;"
      >
        <img
          alt="dzing"
          class="icon-wrapper__icon media_lazy"
          data-src="[object Object]"
          data-testid="customer-icon"
        />
      </div>
      <div
        class="icon-wrapper"
        data-testid="icon-wrapper"
        style="background-color: transparent;"
      >
        <img
          alt="topica"
          class="icon-wrapper__icon media_lazy"
          data-src="[object Object]"
          data-testid="customer-icon"
        />
      </div>
      <div
        class="icon-wrapper"
        data-testid="icon-wrapper"
        style="background-color: transparent;"
      >
        <img
          alt="veeqo"
          class="icon-wrapper__icon media_lazy"
          data-src="[object Object]"
          data-testid="customer-icon"
        />
      </div>
      <div
        class="icon-wrapper"
        data-testid="icon-wrapper"
        style="background-color: transparent;"
      >
        <img
          alt="guardrails"
          class="icon-wrapper__icon media_lazy"
          data-src="[object Object]"
          data-testid="customer-icon"
        />
      </div>
      <div
        class="icon-wrapper"
        data-testid="icon-wrapper"
        style="background-color: transparent;"
      >
        <img
          alt="teacherly"
          class="icon-wrapper__icon media_lazy"
          data-src="[object Object]"
          data-testid="customer-icon"
        />
      </div>
      <div
        class="icon-wrapper"
        data-testid="icon-wrapper"
        style="background-color: transparent;"
      >
        <img
          alt="nambataxi"
          class="icon-wrapper__icon media_lazy"
          data-src="[object Object]"
          data-testid="customer-icon"
        />
      </div>
      <div
        class="icon-wrapper"
        data-testid="icon-wrapper"
        style="background-color: transparent;"
      >
        <img
          alt="alfa-bank"
          class="icon-wrapper__icon media_lazy"
          data-src="[object Object]"
          data-testid="customer-icon"
        />
      </div>
      <div
        class="icon-wrapper"
        data-testid="icon-wrapper"
        style="background-color: transparent;"
      >
        <img
          alt="edumall"
          class="icon-wrapper__icon media_lazy"
          data-src="[object Object]"
          data-testid="customer-icon"
        />
      </div>
      <div
        class="icon-wrapper"
        data-testid="icon-wrapper"
        style="background-color: transparent;"
      >
        <img
          alt="triggermesh"
          class="icon-wrapper__icon media_lazy"
          data-src="[object Object]"
          data-testid="customer-icon"
        />
      </div>
    </div>
     
    <button
      class="ui-link-btn projects-countries_link-btn"
      type="button"
    >
      <nuxtlink
        class="ui-link-btn_link"
        to="/case-studies/"
      >
        
    See all case studies 
  
      </nuxtlink>
    </button>
  </div>
</div>
`;
